### PR TITLE
Fixes an issue with decorators when compiling using tsc.exe

### LINKED ISSUE #5229: False expression: Unsupported decorator target
Hi,

From 1.5 to 1.6 migration, we have a project that doesn't compile anymore and it give us only this tsc error message:
Debug Failure. False expression: Unsupported decorator target.

There is the faillure stack trace I get by adding a ts.sys,write of the stack where the error occur:

    Error: Debug Failure. False expression: Unsupported decorator target.
    at assert (:724:17)
    at fail (:729:13)
    at getEffectiveDecoratorFirstArgumentType (:16779:21)
    at getEffectiveDecoratorArgumentType (:16842:17)
    at getEffectiveArgumentType (:16855:17)
    at inferTypeArguments (:16660:21)
    at chooseOverload (:16985:33)
    at resolveCall (:16919:17)
    at resolveDecorator (:17144:13)
    at getResolvedSignature (:17160:21)
    tsc.js(29826, 17) JavaScript runtime error : Debug Failure. False expression: Unsupported decorator target.

I try to comment partially decorator to find the problem wihout success.

Can't share source to help. Maybe if it help I can share my decorator signatures.

Thanks.

### PR DESCRIPTION
**Pull request by rbuckton**
Switched `switch` statement to `if` branches due to a switch-case optimization issue in tsc.exe

Fixes #5229.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -8646,39 +8646,36 @@ namespace ts {
           */
         function getEffectiveDecoratorFirstArgumentType(node: Node): Type {
             // The first argument to a decorator is its `target`.
-            switch (node.kind) {
-                case SyntaxKind.ClassDeclaration:
-                case SyntaxKind.ClassExpression:
```
Comment (sandersn) [inline src/compiler/checker.ts]: what happens with ClassExpression in the `if` version? it looks like it'll hit the Debug.fail

**Comment by sandersn** [comment]
Did you get a usable repro from #5229? This fix should have an associated test and that seems like a good source for it.

File: src/compiler/checker.ts (hunk 1 above)
Comment (rbuckton) [inline src/compiler/checker.ts]: ClassExpressions aren't supported, that case was incorrect to begin with.

**Comment by rbuckton** [comment]
No, I could not repro this locally but worked with the customer to determine the exact nature of the issue. And found that this fix addresses the problem. Unfortunately, this error is due to a bug in the specific version of `jscript.dll` that we use for `tsc.exe`. Since our tests run in node, we would never see this issue. I've talked with @mhegazy offline about seeing if we can get a backwards compatible, yet more recent drop of our script host for future releases.

**Comment by sandersn** [comment]
OK, makes sense. In that case, :+1:

**Pull request merged**